### DISCUSSION TITLE
Bundle migrations

### DISCUSSION THREAD

**Discussion by GrandChaman** [SeaQL/sea-orm#257] [Q&A] [answered]
Hello!

Is there a way to bundle the migrations file and to apply them when first connecting to the database ?

In the same fashion that `sqlx` does it.

**Comment by tyt2y3** [comment]
https://github.com/SeaQL/sea-orm/blob/349108dad66cae0934df960cbcc4848bdcf4ea5c/examples/rocket_example/src/main.rs#L166-L176

I think the way integration should work will depend on the framework you use.
You can reference the Rocket example for now.
You can, actually, still use `sqlx` migration, if writing schema in SQL is your choice.

**Comment by GrandChaman** [comment]
> I think the way integration should work will depend on the framework you use.

I think it makes sense!

However, it'd may be interesting to allow accessing the inner value of `Sqlx<Something>PoolConnection` so that one could run [sqlx::migrate](https://docs.rs/sqlx/0.5.9/sqlx/macro.migrate.html) on start-up.

**Comment by tyt2y3** [comment, marked as answer]
I think currently you can do it the other way around. You setup a sqlx connection yourself, run the necessary migrations you want, and then convert it into a SeaORM connection using [`from_sqlx_*_pool`](https://docs.rs/sea-orm/0.3.0/sea_orm/?search=from_sqlx)
